hs_geographic_feature_resource/receivers.py

```diff
@@ -212,51 +212,65 @@ def geofeature_pre_create_resource(sender, **kwargs):
     tmp_dir = None
     try:
         files = kwargs['files']
-        metadata = kwargs['metadata']
-        validate_files_dict = kwargs['validate_files']
         fed_res_fnames = kwargs['fed_res_file_names']
         fed_res_path = kwargs['fed_res_path']
-        file_selected = False
-
-        if files:
-            file_selected = True
-            file_list = [file.name for file in files]
-        elif fed_res_fnames:
-            file_list = utils.get_fed_zone_files(fed_res_fnames)
-            if file_list:
-                file_selected = True
-
-        if file_selected:
-            files_type = check_uploaded_files_type(files, file_list)
+        metadata = kwargs['metadata']
+        validate_files_dict = kwargs['validate_files']
+
+        if files and fed_res_fnames:
+            validate_files_dict['are_files_valid'] = False
+            validate_files_dict['message'] = 'Please upload files from either local disk or irods, not both.'
+            return
+
+        if files or fed_res_fnames:
+            files_list = [f for f in files]
```

Comment:
@zhiyuli Even files are already retrieved to local Django server, there is some performance overhead to wrap it into UploadedFile(), see https://docs.djangoproject.com/en/1.9/topics/http/file-uploads/ for details, although if we do it carefully, the performance overhead might be manageable, but the way you are doing currently will definitely fail for large files since you are passing in ```file=open(file_path, 'r')``` as a ```file``` parameter into UploadedFile() initializer, which will try to read big files all together rather than do them in chunks. Let me create another function called ```utils.get_fed_zone_files_into_uploaded_files()``` just to accommodate your user case here to minimize overhead in wrapping it into UploadedFile() object. For other resource types, they are already passing the temp file directly into gdal functions for processing without this extra step to wrap it into UploadedFile() which is the most efficient way without introducing extra performance overhead which may not matter for regular size files but will make a difference for big files. I will make changes accordingly so that your receiver can directly append returned UploadedFile() object for files retrieved from federated zone using the new utility function I will add, but keep in mind in the future if there is a performance issue related to big files for feature resource type, this is one place that performance can be improved.